server/lorax_server/models/flash_causal_lm.py

```diff
@@ -1014,6 +1023,33 @@ def generate_token(
                 next_token_id,
                 next_token_logprob,
         ) in enumerate(iterator):
+            if request.parameters.return_k_alternatives > 0:
+                # Limit the number of alternatives to the vocabulary size
+                num_alternatives = min(request.parameters.return_k_alternatives, len(alternative_token_ids[i]))
+
+                # Select top-k logprobs
+                batch_alternative_token_ids = alternative_token_ids[i][:num_alternatives]
+                batch_alternative_token_logprobs = alternative_token_logprobs[i][:num_alternatives]
+
+                # Decode tokens
+                alternative_token_texts = list()
+                for alternative_token_id in batch_alternative_token_ids:
+                    all_input_ids.append(alternative_token_id)
+                    alternative_token_text, _, _ = self.decode_token(
+                        all_input_ids,
+                        prefix_offset,
+                        read_offset,
+                    )
+                    alternative_token_texts.append(alternative_token_text)
+                    all_input_ids.pop()
+                alternative_tokens = AlternativeTokens(
```

Comment:
It looks like we're overriding this at every iteration of the loop. Should `alternative_tokens` instead be a list of `AlternativeTokens`, one for each request in the batch?

Reply:
We are adding the `alternative_tokens` to the `Generation` that is created for each request, so I don't see a problem with overriding it. Basically I just copied from what is done with the `PrefillTokens`. The variables that go into `AlternativeTokens` are kind of badly named though, so I added another commit to fix that :)